gateway/modules/eventing/routine.go

```diff
@@ -12,28 +14,137 @@ import (
 
 func (m *Module) createProcessUpdateEventsRoutine() {
 	m.updateEventC = make(chan *queueUpdateEvent, 1000)
-	for i := 0; i < 50; i++ {
+	for i := 0; i < 25; i++ {
 		go m.routineProcessUpdateEvents()
 	}
 }
 
 func (m *Module) routineProcessUpdateEvents() {
-	for ev := range m.updateEventC {
-		m.queueUpdateEvent(ev)
+	for {
+		select {
+		case <-m.globalCloserChannel:
+			fmt.Println("Closing routineProcessStaged")
+			return
+		case ev := <-m.updateEventC:
+			m.queueUpdateEvent(ev)
+			m.deleteEventFromProcessingEventsMapChannel <- []string{ev.req.Find["_id"].(string)}
+		}
+	}
+}
+
+func (m *Module) routineUpdateEventsStatusInDB(updateChan chan *queueUpdateEvent) {
+	duration := 1 * time.Second
+	t := time.NewTimer(duration)
+	defer t.Stop()
+	arr := make([]*queueUpdateEvent, 0)
+	for {
+		select {
+		case <-m.globalCloserChannel:
+			close(m.updateFailedEventInDBChannel)
```

Comment:
You can close this channel. Only the person writing to the channel is allowed to close the channel